### PR TITLE
WFCORE-3891: ensure unique names for default formatters

### PR DESCRIPTION
**Pull request by michpetrov**
Issue: [WFCORE-3891](https://issues.jboss.org/browse/WFCORE-3891)

### CODE REVIEW

File: logging/src/main/java/org/jboss/as/logging/formatters/PatternFormatterResourceDefinition.java [hunk 1]
```diff
@@ -111,6 +113,9 @@ public boolean isMarshallable(final AttributeDefinition attribute, final ModelNo
         @Override
         public void performRuntime(final OperationContext context, final ModelNode operation, final ModelNode model, final LogContextConfiguration logContextConfiguration) throws OperationFailedException {
             final String name = context.getCurrentAddressValue();
+            if (name.endsWith(DEFAULT_FORMATTER_SUFFIX)) {
+                throw new IllegalArgumentException("Formatter name cannot end with '" + DEFAULT_FORMATTER_SUFFIX + "'");
```
Comment (jamezp) [inline logging/src/main/java/org/jboss/as/logging/formatters/PatternFormatterResourceDefinition.java]: This needs to be internationalized and I think an `OperationFailedException` would be a better one to throw.

File: logging/src/main/java/org/jboss/as/logging/formatters/StructuredFormatterResourceDefinition.java [hunk 2]
```diff
@@ -369,6 +369,9 @@ public void performRuntime(final OperationContext context, final ModelNode opera
             }
 
             final String name = context.getCurrentAddressValue();
+            if (name.endsWith(PatternFormatterResourceDefinition.DEFAULT_FORMATTER_SUFFIX)) {
+                throw new IllegalArgumentException("Formatter name cannot end with '" + PatternFormatterResourceDefinition.DEFAULT_FORMATTER_SUFFIX + "'");
```
Comment (jamezp) [inline logging/src/main/java/org/jboss/as/logging/formatters/StructuredFormatterResourceDefinition.java]: This needs to be internationalized and I think an `OperationFailedException` would be a better one to throw.

File: logging/src/main/java/org/jboss/as/logging/handlers/HandlerOperations.java [hunk 3]
```diff
@@ -650,22 +651,19 @@ private static void handleProperty(final AttributeDefinition attribute, final Op
             if (valueNode.isDefined()) {
                 final String resolvedValue = valueNode.asString();
                 configuration.setFormatterName(resolvedValue);
-                // If the formatter was previously defined by the formatter attribute, remove the formatter
-                if (logContextConfiguration.getFormatterNames().contains(handlerName)) {
-                    logContextConfiguration.removeFormatterConfiguration(handlerName);
-                }
```
Comment (jamezp) [inline logging/src/main/java/org/jboss/as/logging/handlers/HandlerOperations.java]: We still need this to remove the `handlerName + PatternFormatterResourceDefinition.DEFAULT_FORMATTER_SUFFIX`.

File: logging/src/main/java/org/jboss/as/logging/formatters/PatternFormatterResourceDefinition.java (hunk 1 above)
Comment (michpetrov) [reply logging/src/main/java/org/jboss/as/logging/formatters/PatternFormatterResourceDefinition.java]: How do I do that? Should I turn this into a new message in LoggingLogger?

File: logging/src/main/java/org/jboss/as/logging/handlers/HandlerOperations.java (hunk 3 above)
Comment (michpetrov) [reply logging/src/main/java/org/jboss/as/logging/handlers/HandlerOperations.java]: That is taken care of [here](https://github.com/wildfly/wildfly-core/blob/8839ff61c041f930b4128d1d7351d779a8981159/logging/src/main/java/org/jboss/as/logging/handlers/HandlerOperations.java#L632), isn't it? The test is passing.

File: logging/src/main/java/org/jboss/as/logging/formatters/PatternFormatterResourceDefinition.java (hunk 1 above)
Comment (jamezp) [reply logging/src/main/java/org/jboss/as/logging/formatters/PatternFormatterResourceDefinition.java]: Yes. You add a method to the `LoggingLogger` that returns an `OperationFailedException`.

File: logging/src/main/java/org/jboss/as/logging/handlers/HandlerOperations.java (hunk 3 above)
Comment (jamezp) [reply logging/src/main/java/org/jboss/as/logging/handlers/HandlerOperations.java]: Only if the `formatter` attribute is processed. So for example if you were to add a handler then write a `named-formatter` attribute, that remove would not be hit.

For example:
```
/subsystem=logging/console-handler=my-console:add
/subsystem=logging/console-handler=my-console:write-attribute(name=named-formatter, value=COLOR-PATTERN)
```

**Pull request merged**